### PR TITLE
Fix move_to event issue on macOS (#19)

### PR DESCRIPTION
**Pull request by bwpge**
Tested on macOS Monterey 12.6.3, new implementation correctly fires events that register in open applications. All tests pass with `cargo test`.

**Breaking Changes**

- (macOS) Changed `Mouse::move_to` behavior
    - The `x` and `y` coordinates are now treated as global coordinates instead of display coordinates
    - This is a byproduct of changing from [`CGDisplayMoveCursorToPoint`](https://developer.apple.com/documentation/coregraphics/1455258-cgdisplaymovecursortopoint) to [`CGEventCreateMouseEvent`](https://developer.apple.com/documentation/coregraphics/1454356-cgeventcreatemouseevent)
    - New behavior is more in line with Windows implementation (see: [`SetCursorPos`](https://learn.microsoft.com/en-us/windows/win32/api/winuser/nf-winuser-setcursorpos))

**Non-breaking Changes**

- Removed `Error::CGDisplayMoveCursorToPoint(CGError)` variant (no longer used by the library and not part of the public API)
- Removed unused imports in `sys::macos`

### CODE REVIEW

**Comment by bwpge** [comment]
Apologies for the comment spam. I forgot to mention in the above notes:

The new implementation requires the user to allow "Accessibility" control to the calling application (vscode, iTerm, Terminal, etc.) when moving the mouse. This shouldn't be that big of a deal, since the `Mouse::press` already required those permissions. However, it is conceivable that someone could be **only** using this library to move the mouse around. In that use case, they were never prompted to allow the application to take control (the `CGDisplayMoveCursorToPoint` function does not require this level of access on macOS).

This could be a potentially jarring and off-putting result for that particular subset of library consumers -- e.g., they were never prompted to allow this access, and then when they update their dependency the application is asking to take control of their machine.

Perhaps we would want to note this behavior in the README since macOS loves their scary messages and prompts. Happy to submit a PR for that.

**Comment by AltF02** [comment]
Yeah I agree, I was unaware this was a required option. Thanks for all your work!

File: src/sys/macos.rs [hunk 1]
```diff
@@ -66,10 +61,16 @@ impl Mouse {
...
-        match unsafe { CGDisplayMoveCursorToPoint(CGMainDisplayID(), point) } {
-            0 => Ok(()),
-            err => Err(Box::new(Error::CGDisplayMoveCursorToPoint(err))),
-        }
+        CGEvent::new_mouse_event(
+            Self::event_source()?,
+            CGEventType::MouseMoved,
+            point,
+            CGMouseButton::Left, // ignored
+        )
```
Comment (AltF02) [inline src/sys/macos.rs:69]: Disclaimer: I dont have any idea how core graphics work in MacOS

Is there a reason why this parameter is required for a move?

Comment (bwpge) [reply src/sys/macos.rs:69]: Fair question, definitely not an expert myself either :)

With respect to the `core_graphics` crate, [`CGEvent::new_mouse_event`](https://docs.rs/core-graphics/latest/core_graphics) requires this argument. It's basically just a wrapper call to the private function [`CGEventCreateMouseEvent`](https://docs.rs/core-graphics/latest/src/core_graphics/event.rs.html#711-712), which notes "`mouseButton` is ignored unless `mouseType` is one of `kCGEventOtherMouseDown`, `kCGEventOtherMouseDragged`, or `kCGEventOtherMouseUp`."

From the Apple docs, this appears to be intentional: https://developer.apple.com/documentation/coregraphics/1454356-cgeventcreatemouseevent (see `mouseButton` description).

**Comment by bwpge** [comment]
Just checking in if this is ok to merge. Happy to fix any outstanding issues :)

**Comment by AltF02** [comment]
Hi sorry, I've been really busy these past few days

**Review by AltF02 — APPROVED** [review]

**Comment by bwpge** [comment]
No worries, thanks for the quick reply!

**Pull request merged**